js/api.js

```diff
@@ -0,0 +1,72 @@
+const base_url = 'https://js-todo-list-9ca3a.df.r.appspot.com';
+
+const option = {
+    post: (contents) => ({
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(contents),
+    }),
+    
+    put: (contents) => ({
+        method: 'PUT',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(contents),
+    }),
+
+    delete: () => ({
+        method: 'DELETE',
+    }),
+
+};
+
+const request = async (url, option = {}) => {
+    try{
+        const response = await fetch(`${base_url}${url}`, option);
+        if(!response.ok){
+            throw new Error(response.status);
+        }
+        return response.json();
+    } catch(err){
+        alert(`Error: ${err}`);
+    }
+};
+
+export const API = {
+    addUser: (userName) => {
+        const content = {
+            name: userName,
+        };
+        return request('/api/users', option.post(content));
+    },
```

Comment:
```suggestion
    addUser(userName) {
        const content = {
            name: userName,
        };
        return request('/api/users', option.post(content));
    },
```
es6부터 객체 내에서 함수를 프로퍼티로 전달할때, 위와 같이 쓸수 있습니다!